src/Pim/Component/Api/Normalizer/Exception/ViolationNormalizer.php

```diff
@@ -137,14 +142,64 @@ protected function getProductValuesErrors(ConstraintViolationInterface $violatio
             'message'   => $violation->getMessage(),
             'attribute' => $productValue->getAttribute()->getCode(),
             'locale'    => $productValue->getLocale(),
-            'scope'     => $productValue->getScope()
+            'scope'     => $productValue->getScope(),
         ];
 
         if (AttributeTypes::PRICE_COLLECTION === $attributeType &&
-            null !== $violation->getInvalidValue()->getCurrency()) {
+            null !== $violation->getInvalidValue()->getCurrency()
+        ) {
             $error['currency'] = $violation->getInvalidValue()->getCurrency();
         }
 
         return $error;
     }
+
+    /**
+     * This function filters the constraints regarding the identifier size of the identifier.
+     *
+     * The product field "identifier" introduced during the single storage development (in addition to the "identifier"
+     * product value) added a new Length constraint on this property (see the product entity mapping in doctrine)
+     * which is breaking the API.
+     *
+     * This method aims to filter this new constraint and to normalize only the constraint regarding the product value
+     * (Because its Length max number is dynamic compared to the identifier property).
+     *
+     * TODO: To remove once the "identifier" product value is removed from the product value collection.
+     *
+     * @param ConstraintViolationListInterface $violations
+     *
+     * @return array
+     */
+    protected function filterLengthConstraintIdentifierViolations(ConstraintViolationListInterface $violations)
+    {
+        $filteredViolations = [];
+        $identifierViolations = [];
+
+        foreach ($violations as $violation) {
+            if ($violation->getRoot() instanceof ProductInterface &&
```

Comment:
this method is already a "patapouille" for a problem we have
this one problem is isolated inside one method that is very well documented

I don't see the point of splitting it into 2 different methods
this will be imo counter productive and worse that the current code